From f2198ddbac7d420b8080d4c69fa62700de10b2fd Mon Sep 17 00:00:00 2001
From: Wei Li <liwei391@huawei.com>
Date: Tue, 6 Jul 2021 16:50:25 +0800
Subject: [PATCH] locking/qspinlock: Disable CNA by default
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: f2198ddbac7d420b8080d4c69fa62700de10b2fd


hulk inclusion
category: feature
bugzilla: 169576
CVE: NA

-------------------------------------------------

Disable CNA by default, this default behavior can be overridden with
the kernel boot command-line option "numa_spinlock=on/off/auto".

Signed-off-by: Wei Li <liwei391@huawei.com>
Reviewed-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/locking/qspinlock_cna.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/locking/qspinlock_cna.h b/kernel/locking/qspinlock_cna.h
index 17d56c739e57..3983505c1118 100644
--- a/kernel/locking/qspinlock_cna.h
+++ b/kernel/locking/qspinlock_cna.h
@@ -378,9 +378,9 @@ static inline void cna_lock_handoff(struct mcs_spinlock *node,
 
 /*
  * Constant (boot-param configurable) flag selecting the NUMA-aware variant
- * of spinlock.  Possible values: -1 (off) / 0 (auto, default) / 1 (on).
+ * of spinlock.  Possible values: -1 (off, default) / 0 (auto) / 1 (on).
  */
-static int numa_spinlock_flag;
+static int numa_spinlock_flag = -1;
 
 static int __init numa_spinlock_setup(char *str)
 {
-- 
2.26.2

